src/test/clojure/clara/test_durability.clj

```diff
@@ -254,4 +254,24 @@
             (check-fact expected-fact fact)))))))
  
 (deftest test-durability-fressian-serde
-  (durability-test :fressian))
+  (testing "SerDe of the rulebase along with working memory"
+    (durability-test :fressian))
+
+  (testing "Repeated SerDe of rulebase"
+    (let [rb-serde (fn [s]
+                     (with-open [baos (java.io.ByteArrayOutputStream.)]
+                       (d/serialize-rulebase s (df/create-session-serializer baos))
+                       (let [rb-data (.toByteArray baos)]
+                         (with-open [bais (java.io.ByteArrayInputStream. rb-data)]
+                           (d/deserialize-rulebase (df/create-session-serializer bais))))))
+
+          s (mk-session 'clara.durability-rules)
+          rb (-> s eng/components :rulebase)
+          deserialized1 (rb-serde s)
+          ;; Need a session to do the 2nd round of SerDe.
+          restored (d/assemble-restored-session deserialized1 {})
+          deserialized2 (rb-serde restored)]
+      
+      (is (= (-> rb :id-to-node keys set)
+             (-> deserialized1 :id-to-node keys set)
+             (-> deserialized2 :id-to-node keys set))))))
```

Comment:
Yeah I just did this to get a quick test.  It wasn't meant to be much more than asserting that something structurally coherent can serialize and deserialize.  I know the `:id-to-node` doesn't really prove much of anything in terms of the correctness of a rulebase.  

Prior to these changes there were exceptions before even getting this far.  I can try to work in a test about inserting facts to these rulebases and showing they all have the same queried results.